tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,755 @@
+    for output in SAMPLE_OUTPUTS_ON:
+        mock_api_object.change_output.assert_any_call(
+            output["id"], selected=output["selected"], volume=output["volume"],
+        )
+    mock_api_object.set_volume.assert_any_call(volume=0)
+    mock_api_object.set_volume.assert_any_call(volume=SAMPLE_PLAYER_PAUSED["volume"])
+    mock_api_object.set_volume.assert_any_call(volume=50)
+    mock_api_object.set_enabled_outputs.assert_any_call(
+        [output["id"] for output in SAMPLE_OUTPUTS_ON]
+    )
+    mock_api_object.set_enabled_outputs.assert_any_call([])
+    mock_api_object.start_playback.assert_called_once()
+    assert mock_api_object.pause_playback.call_count == 3
+    mock_api_object.stop_playback.assert_called_once()
+    mock_api_object.previous_track.assert_called_once()
+    mock_api_object.next_track.assert_called_once()
+    mock_api_object.seek.assert_called_once()
+    mock_api_object.shuffle.assert_called_once()
+    mock_api_object.clear_queue.assert_called_once()
+
+
+async def test_async_play_media_from_paused(hass, mock_api_object):
+    """Test async play media from paused."""
+    initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+            ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.state == initial_state.state
+    assert state.last_updated > initial_state.last_updated
+
+
+async def test_async_play_media_from_stopped(
+    hass, get_request_return_values, mock_api_object
+):
+    """Test async play media from stopped."""
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+
+    get_request_return_values["player"] = SAMPLE_PLAYER_STOPPED
+    await updater_update(["player"])
+    await hass.async_block_till_done()
+    initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+            ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.state == initial_state.state
+    assert state.last_updated > initial_state.last_updated
+
+
+async def test_async_play_media_unsupported(hass, mock_api_object):
+    """Test async play media on unsupported media type."""
+    initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_TVSHOW,
+            ATTR_MEDIA_CONTENT_ID: "wontwork.mp4",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.last_updated == initial_state.last_updated
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)